### ISSUE TITLE
[CS0652] Add range check for "Math.Clamp" 

### ISSUE THREAD

**Issue by TobiasSekan** [dotnet/roslyn#48004]
Hello,

**Brief description:**

I would be nice when Roslyn can check min and max range of a value when it is used in `Math.Clamp` and report **CS0652**

**Languages applicable:**
C# 9.0, .NET 3.1

**Code example that the analyzer should report:**
```cs
/// <summary>
/// Save convert a signed integer value to a unsigned integer value
/// </summary>
/// <param name="value">The singed integer value to convert</param>
/// <returns>A singed integer value or <see cref="uint.MinValue"/> when the given value was lower then <see ref="uint.MinValue"/></returns>
internal static uint SaveConvertToUint(this int value)
{
    return Convert.ToUInt32(Math.Clamp(value, uint.MinValue, uint.MaxValue));
}  
```

**Additional information:**
The analyzer currently report **CS0652** when compare direct
```cs
internal static uint SaveConvertToUint(this int value)
{
    if(value < uint.MinValue)
    {
        return uint.MinValue;
    }

    // CS0652: Comparison to integral constant is useless; the constant is outside the range of type 'int'
    if(value > uint.MaxValue)
    {
        return uint.MaxValue;
    }

    return Convert.ToUInt32(value);
}
```

Any more additional information you would like to add.

**Comment by Youssef1313** [comment]
@TobbiasSekan A small clarification, CS0652 is a **compiler** warning, not an analyzer.

I think your suggestion should be a new analyzer.

@mavasani Since the suggestion here isn't about code style, should it belong to roslyn-analyzers?

**Comment by TobiasSekan** [comment]
> 
> @TobbiasSekan A small clarification, CS0652 is a **compiler** warning, not an analyzer.
> 

Good to know 👍

**Comment by mavasani** [comment]
This seems a very specific suggestion to enhance the compiler diagnostic `CS0652` or introduce a new compiler warning (as part of compiler warning wave). Moving it to Roslyn.

**Comment by mavasani** [comment]
Tagging @333fred

**Comment by 333fred** [comment]
@mavasani I don't believe this is something we can implement in the compiler. We don't know anything about `Math.Clamp` today, and since there are no language features that depend on knowledge of that method, the appropriate place for such a warning would be in an analyzer, not in the compiler. As such, I'm going to transfer this.
